hermes/probe/monitor.go

```diff
@@ -12,9 +12,164 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 //
+// Authors:
+// - Alicja Kwiecinska (kwiecinskaa@google.com) github: alicjakwie
+// - Evan Spendlove, GitHub: evanSpendlove.
+//
 // Monitor implements a probe that monitors a storage system using the
 // Hermes algorithm.
+// Package probe implements the probe that Hermes uses to monitor
+// a storage system.
+
+package probe
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/google/cloudprober/logger"
+	"github.com/google/cloudprober/metrics"
+	"github.com/google/cloudprober/probes/options"
+	"github.com/googleinterns/step224-2020/hermes"
+
+	probepb "github.com/googleinterns/step224-2020/config/proto"
+	hermespb "github.com/googleinterns/step224-2020/hermes/proto"
+)
 
 // MonitorProbe holds aggregate information about all probe runs, per-target.
 // It also holds the config and options used to initialise the probe.
-type MonitorProbe struct{}
+type MonitorProbe struct {
+	name    string
+	Config  *probepb.HermesProbeDef
+	targets []*probepb.Target
+	opts    *options.Options
```

Comment:
What is the difference between config & opts, and could you combine them?

Reply:
Config contains the proto configuration of the probe.
Opts contains various values and variables pulled/generated from the config.

Both are required for integration with Cloudprober. 

Even though we don't use opts internally, we need it as Cloudprober will use it.

Reply:
I agree that as a reader I'm puzzled by the presence of both "config" and "opts".